cmd/geth/chaincmd.go

```diff
@@ -675,8 +711,202 @@ func dump(ctx *cli.Context) error {
 	return nil
 }
 
+func exportSegment(ctx *cli.Context) error {
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	db := utils.MakeChainDatabase(ctx, stack, false, false)
+	defer db.Close()
+
+	genesisHash := rawdb.ReadCanonicalHash(db, 0)
+	td := rawdb.ReadTd(db, genesisHash, 0)
+	chainConfig, engine, headerChain, err := simpleHeaderChain(db, genesisHash)
+	if err != nil {
+		return err
+	}
+	latest := headerChain.CurrentHeader()
+	if _, ok := engine.(consensus.PoSA); !ok {
+		return errors.New("cannot generate history segment because consensus engine is not PoSA")
+	}
+	if !chainConfig.IsLuban(latest.Number) {
+		return errors.New("luban hard fork is not enabled , cannot generate history segment")
+	}
+
+	var (
+		boundStartBlock      = params.BoundStartBlock
+		historySegmentLength = params.HistorySegmentLength
+	)
+	switch genesisHash {
+	case params.BSCGenesisHash, params.ChapelGenesisHash, params.RialtoGenesisHash:
```

Comment:
to test locally, should support flag RialtoHash
https://github.com/bnb-chain/bsc/commit/a6befb5078425c148eb74943586e4b438264af78